src/Math-Tests-Matrix/PMMatrixTest.class.st

```diff
@@ -737,6 +737,29 @@ PMMatrixTest >> testSkalarMultiplication [
 	self assert: r class equals: PMSymmetricMatrix
 ]
 
+{ #category : #tests }
+PMMatrixTest >> testStrassenProductWithMatrix [
+		
+	| aPMMatrix expected strassenProduct |
+	
+	aPMMatrix := PMMatrix rows: #( (1 2 3 4) (1 0 1 2) (2 3 1 4) (4 3 2 2)).
+	expected  := PMMatrix rows: #( (30 12 27 24) (12 6 11 10) (27 11 30 27) (24 10 27 33)).
+	strassenProduct := aPMMatrix transpose strassenProductWithMatrix: aPMMatrix.
+	self assert: strassenProduct equals: expected.
+]
+
+{ #category : #tests }
+PMMatrixTest >> testStrassenProductWithMatrixWithInvalidShapes [
+		
+	| aPMMatrix |
+	
+	"all the dimension of the matrices should be a power of 2 for strassen multiplication"
+	aPMMatrix := PMMatrix rows: #( (1 2 3) (1 0 1) (2 3 1) (2 0 1)).
+	
+	self should: [ aPMMatrix strassenProductWithMatrix: aPMMatrix transpose ] raise: AssertionFailure.
```

Comment:
@rakki-18 I just thought of something that maybe worth experimenting with. With the `assert: description:` message, is there a way to have it raise a more specific exception, such as `MatrixHasInvalidShape` - this helps with debugging, but has the disadvantage of YAGNI, unless the kind of shape checking done here is also done elsewhere, meaning a possible refactoring opportunity.

Reply:
@hemalvarambhia We could try making a `MatrixHasInvalidShape` exception class. But as you mentioned, I could only find "matrix multiplication" where this class could be potentially used.

Reply:
Explore the idea, if you can, then, to examine the consequences for the design and debugging

Reply:
Yeah, I will try to explore.